### PR TITLE
feat: report spam NFTs

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Adds a "Report Spam" button marking the collection as spam and hiding it in the NFT slice, as well as reporting the NFT as spam to NFT, as long as the NFT is from a supported chain for the `reportSpam` endpoint: https://docs.alchemy.com/reference/reportspam i.e Ethereum, Mainnet, and Polygon.

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes https://github.com/shapeshift/web/issues/5075

## Risk

Theoretically none - worst that happens, we fire a malformed request to Alchemy or don't fire a request to report the spammy NFT collection when we should, though tested and this does what it says on the box.

## Testing

- Click the "Report Spam" button
- Ensure a network request is made to Alchemy, and a log with the response from Alchemy appears in the console
- Ensure the modal closes after the report, and the collection NFT/s is/are now gone

### Engineering

- ☝🏽 

### Operations

- ☝🏽 

## Screenshots (if applicable)

- Develop

image
image
image

- NFT reporting

image

- This diff - all spam NFTs reported

image
image

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #5276** <a href="https://app.graphite.dev/github/pr/shapeshift/web/5276" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/5276?utm_source=stack-comment).

File: src/components/Modals/Nfts/NftModal.tsx [hunk 1]
```diff
@@ -138,6 +140,47 @@ export const NftModal: React.FC<NftModalProps> = ({ nftAssetId }) => {
...
+            getConfig().REACT_APP_ALCHEMY_POLYGON_JAYPEGS_API_KEY
+          }`
+        case ethChainId:
+          return `https://eth-mainnet.g.alchemy.com/nft/v2/${
+            getConfig().REACT_APP_ALCHEMY_ETHEREUM_JAYPEGS_API_KEY
+          }`
+        default:
+          return undefined
+      }
+    })()
```
Comment (gomesalexandre) [inline src/components/Modals/Nfts/NftModal.tsx:163]: Will need to be made env base URLs

File: src/components/Nfts/NftTable.tsx [hunk 2]
```diff
@@ -38,7 +38,9 @@ export const NftTable = () => {
   const [networkFilters, setNetworkFilters] = useState<ChainId[]>([])
 
   const isLoading = useAppSelector(selectGetNftUserTokensPending)
-  const nftItems = useAppSelector(selectPortfolioNftItemsWithCollection)
+  const nftItems = useAppSelector(selectPortfolioNftItemsWithCollectionExcludeSpams)
+
+  console.log({ nftItems })
```
Comment (gomesalexandre) [inline src/components/Nfts/NftTable.tsx:43]: Remove when opening me

File: src/components/Modals/Nfts/NftModal.tsx (hunk 1 above)
Comment (gomesalexandre) [reply src/components/Modals/Nfts/NftModal.tsx]: 5bcb3db859ad382da66b5195c44a15cc041668f

File: src/components/Modals/Nfts/NftModal.tsx [hunk 3]
```diff
@@ -138,6 +140,47 @@ export const NftModal: React.FC<NftModalProps> = ({ nftAssetId }) => {
     dispatch(nftApi.endpoints.getNft.initiate({ assetId: nftAssetId }, { forceRefetch: true }))
   }, [dispatch, nftAssetId])
 
+  const handleReportSpamClick = useCallback(async () => {
+    // Note, we're using collectionId vs. nftAssetId since the NFT AssetId contains <address>/<tokenId>
+    // and deserializing the asset would give us an assetReference in that form, which is what we want for linking purposes
+    // but not here
```
[the remark this answers is not in the document]
Comment (gomesalexandre) [reply src/components/Modals/Nfts/NftModal.tsx]: 5cb516ed367d253a298ca23cb738e9779f328a84

File: src/components/Nfts/NftTable.tsx (hunk 2 above)
Comment (gomesalexandre) [reply src/components/Nfts/NftTable.tsx]: 676ce2a6

**Review by 0xApotheosis — APPROVED** [review]
Does what it says on the box!

Hopefully those reports start to filter back down server-side so they magically persist on cache clear.

**Pull request merged**